bin/propolis-server/src/lib/spec/api_spec_v0.rs

```diff
@@ -0,0 +1,259 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Conversions from version-0 instance specs in the [`propolis_api_types`]
+//! crate to the internal [`super::Spec`] representation.
+
+use propolis_api_types::instance_spec::{
+    components::devices::{SerialPort as SerialPortDesc, SerialPortNumber},
+    v0::{InstanceSpecV0, NetworkBackendV0, NetworkDeviceV0, StorageDeviceV0},
+};
+use thiserror::Error;
+
+#[cfg(feature = "falcon")]
+use propolis_api_types::instance_spec::components::devices::SoftNpuPort as SoftNpuPortSpec;
+
+#[cfg(feature = "falcon")]
+use crate::spec::SoftNpuPort;
+
+use super::{
+    builder::{SpecBuilder, SpecBuilderError},
+    Disk, Nic, QemuPvpanic, SerialPortUser, Spec,
+};
+
+#[derive(Debug, Error)]
+pub(crate) enum ApiSpecParseError {
```

Comment:
suuuuuuper tiny, unimportant nit: most of these variants don't _really_ seem like "parse errors" --- referencing a backend that doesn't exist isn't a syntactic error in the spec, it's a semantic error...I don't know if this actually matters and i'm probably just being pedantic...